CONTRIBUTING.md

```diff
@@ -40,6 +40,57 @@ please ensure that:
    deprecation, removal etc) that you have added a reno release note for that
    change and that the PR is tagged for the changelog.
 
+### Pull request review, CI, and merge queue
+
+After you've submitted a pull request to Qiskit it will need to pass CI and be
+reviewed by an approved code owner. CI runs get triggered automatically when
+your pull request is opened and on every subsequent commit made to your pull
+request's branch. Code review however may take some time, sometimes even weeks
+or months, there are many new pull requests opened every day and limited number
+of reviewers available, and while every proposed change is a valuable addition
+to the project not everything is the highest priority. You can help this
+process move more quickly by actively reviewing other open PRs. While only
+members of the @Qiskit/terra-core group or listed contributors in CODEOWNERS
+file have permission to provide final approval and mark a PR as ready for
+merging reviewing code is open to everyone and all reviews are welcome and
+extremely valued contributions. Helping with code review also helps reduce
+the burden on the core team and CODEOWNERs to enable them to review more
+code faster.
+
+The code review process is a bit of back and forth where you will receive
+feedback and questions about your proposed changes to the project. You will
+likely have multiple rounds of feedback with suggestions or changes requested
+before approval. Please do not get discouraged as this is normal and part of
+ensuring the quality of the Qiskit project and even what first appears as a
+straightforward or simple change might have larger implications that aren't
+obvious at first. If you receive feedback feel free to request re-review from
+reviewers after you've adjusted your PR based on the comments received.
+
+Another thing to keep in mind is that CI time is a constrained resource and not
+infinite. While waiting for review and approval it is not necessary to keep your
+PR branch up to date on every change to the `main` branch. Doing it periodically
+is fine to make sure there are no regressions as the codebase changes, but doing
```

Comment:
maybe a time suggestion on how often is okey?

Reply:
I thought about that, I was thinking once per week. But a lot of it really depends more on the current ci system backlog than anything else. Like if CI is idle it'd be totally fine to hit the update branch button more frequently, but if there are a bunch of jobs queued it does more harm than good.